bugbot/crash/analyzer.py

```diff
@@ -0,0 +1,710 @@
+            "_facets_size": 10000,
+        }
+
+        def handler(search_results: dict, data: dict):
+            data["num_total_crashes"] = search_results["total"]
+            data["signatures"] = search_results["facets"]["signature"]
+
+        data: dict = {}
+        socorro.SuperSearchUnredacted(
+            params=params,
+            handler=handler,
+            handlerdata=data,
+        ).wait()
+
+        logger.debug(
+            "Fetch info from Socorro for %d signatures", len(data["signatures"])
+        )
+
+        return data["signatures"], data["num_total_crashes"]
+
+    def fetch_bugs(self, include_fields: list[str] = None) -> dict[str, list[dict]]:
+        """Fetch bugs that are filed against the given signatures."""
+
+        params_base: dict = {
+            "include_fields": [
+                "cf_crash_signature",
+            ],
+        }
+
+        if include_fields:
+            params_base["include_fields"].extend(include_fields)
+
+        params_list = []
+        for signatures_chunk in Connection.chunks(list(self._signatures), 30):
+            params = params_base.copy()
+            n = int(utils.get_last_field_num(params))
+            params[f"f{n}"] = "OP"
+            params[f"j{n}"] = "OR"
+            for signature in signatures_chunk:
+                n += 1
+                params[f"f{n}"] = "cf_crash_signature"
+                params[f"o{n}"] = "regexp"
+                params[f"v{n}"] = rf"\[(@ |@){re.escape(signature)}( \]|\])"
+            params[f"f{n+1}"] = "CP"
+            params_list.append(params)
+
+        signatures_bugs: dict = defaultdict(list)
+
+        def handler(res, data):
+            for bug in res["bugs"]:
+                for signature in utils.get_signatures(bug["cf_crash_signature"]):
+                    if signature in self._signatures:
+                        data[signature].append(bug)
+
+        Bugzilla(
+            queries=[
+                connection.Query(Bugzilla.API_URL, params, handler, signatures_bugs)
+                for params in params_list
+            ],
+        ).wait()
+
+        # TODO: remove the call to DevBugzilla after moving to production
+        DevBugzilla(
+            queries=[
+                connection.Query(DevBugzilla.API_URL, params, handler, signatures_bugs)
+                for params in params_list
+            ],
+        ).wait()
+
+        logger.debug(
+            "Total of %d signatures already have bugs filed", len(signatures_bugs)
+        )
+
+        return signatures_bugs
+
+    def analyze(self) -> list[SignatureAnalyzer]:
+        """Analyze the data related to the signatures."""
+        bugs = self.fetch_bugs()
+        # TODO(investigate): For now, we are ignoring signatures that have bugs
+        # filed even if they are closed long time ago. We should investigate
+        # whether we should include the ones with closed bugs. For example, if
+        # the bug was closed as Fixed years ago.
```

Comment:
I agree we should only avoid opening a new bug if a recent bug for the same crash exists or if a old bug exists and is still open

Reply:
I filed an issue to follow up: https://github.com/mozilla/bugbot/issues/2130